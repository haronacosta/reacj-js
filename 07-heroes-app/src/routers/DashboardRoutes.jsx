import { Routes, Route } from 'react-router-dom';

import { Navbar } from '../components/ui/Navbar';
import MarvelScreen from '../components/marvel/MarvelScreen';
import DCScreen from '../components/dc/DCScreen';
import SearchScreen from '../components/search/SearchScreen';
import HeroScreen from '../components/hero/HeroScreen';

const DashboardRoutes = () => {
  return (
    <>
      <Navbar />

      <div className='container'>
        <Routes>
          <Route path='marvel' element={<MarvelScreen />} />
          <Route path='dc' element={<DCScreen />} />
          <Route path='hero/:id' element={<HeroScreen />} />
          <Route path='search' element={<SearchScreen />} />

          <Route path='/' element={<MarvelScreen />} />
        </Routes>
      </div>
    </>
  );
};

export default DashboardRoutes;
