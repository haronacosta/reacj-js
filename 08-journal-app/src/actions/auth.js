import Swal from 'sweetalert2/dist/sweetalert2.js';

import 'sweetalert2/src/sweetalert2.scss';

import { firebase, googleAuthProvider } from '../firebase/firebase-config';
import { types } from '../types/types';
import { deleteNotes } from './notes';
import { uiFinishLoading, uiStartLoading } from './ui';

export const startLoginWithEmailAndPassword = (email, password) => {
  return (dispatch) => {
    dispatch(uiStartLoading());
    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(async ({ user }) => {
        await dispatch(login(user.uid, user.displayName));
        dispatch(uiFinishLoading());
      })
      .catch((err) => {
        dispatch(uiFinishLoading());
        Swal.fire({
          title: 'Error',
          text: err.message,
          icon: 'error'
        });
      });
  };
};

export const startLoginWithGoogle = () => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));
      })
      .catch((error) => {
        Swal.fire({
          title: 'Error',
          text: err.message,
          icon: 'error'
        });
      });
  };
};

export const startRegisterWithEmailPasswordName = (email, password, name) => {
  return (dispatch) => {
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async ({ user }) => {
        await user.updateProfile({ displayName: name });
        dispatch(login(user.uid, user.displayName));
      });
  };
};

export const login = (uid, displayName) => ({
  type: types.login,
  payload: { uid, displayName }
});

export const startLogout = () => {
  return async (dispatch) => {
    await firebase.auth().signOut();

    dispatch(logout());

    dispatch(deleteNotes());
  };
};

export const logout = () => ({
  type: types.logout
});
