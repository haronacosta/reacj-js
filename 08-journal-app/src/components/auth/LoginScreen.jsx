import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { startLoginWithEmailAndPassword, startLoginWithGoogle } from '../../actions/auth';
import { useForm } from '../../hooks/useForm';

const LoginScreen = () => {
  const dispatch = useDispatch();
  const { isLoading } = useSelector((state) => state.ui);

  const [formValues, handleInputChange] = useForm({
    email: '',
    password: ''
  });

  const { email, password } = formValues;

  const handleLogin = (e) => {
    e.preventDefault();
    dispatch(startLoginWithEmailAndPassword(email, password));
  };

  const handleGoogleLogin = () => {
    dispatch(startLoginWithGoogle());
  };

  return (
    <>
      <h3 className="auth__title">Login</h3>

      <form action="post" onSubmit={handleLogin}>
        <input
          className="auth__input"
          type="text"
          placeholder="Your email"
          name="email"
          autoComplete="off"
          value={email}
          onChange={handleInputChange}
        />

        <input
          className="auth__input"
          type="password"
          name="password"
          placeholder="Your password"
          autoComplete="off"
          value={password}
          onChange={handleInputChange}
        />
        <button className="btn btn-primary btn-block" disabled={isLoading} type="submit">
          Login
        </button>
        <hr />

        <div className="auth__social-networks">
          <p>Login with social network</p>
          <div className="google-btn" onClick={handleGoogleLogin}>
            <div className="google-icon-wrapper">
              <img
                className="google-icon"
                src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
                alt="Google button"
              />
            </div>
            <p className="btn-text">
              <b className="btn">Sign in with google</b>
            </p>
          </div>
        </div>

        <Link className="link" to="/auth/register">
          Create new account
        </Link>
      </form>
    </>
  );
};

export default LoginScreen;
