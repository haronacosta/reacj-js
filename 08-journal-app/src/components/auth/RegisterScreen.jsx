import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import validator from 'validator';
import { startRegisterWithEmailPasswordName } from '../../actions/auth';
import { removeUIError, setUIError } from '../../actions/ui';

import { useForm } from '../../hooks/useForm';
const RegisterScreen = () => {
  const dispatch = useDispatch();

  const { msgError } = useSelector((state) => state.ui);

  const [values, handleInputChange] = useForm({
    userName: '',
    email: '',
    password: '',
    confirmPassword: ''
  });

  const { userName, email, password, confirmPassword } = values;

  const handleRegister = (e) => {
    e.preventDefault();

    if (isFormValid()) {
      dispatch(startRegisterWithEmailPasswordName(email, password, userName));
    }
  };

  const isFormValid = () => {
    if (userName.trim().length === 0) {
      dispatch(setUIError('User name is required'));
      return false;
    } else if (!validator.isEmail(email)) {
      dispatch(setUIError('Email is invalid'));
      return false;
    } else if (password !== confirmPassword || password.length < 6) {
      dispatch(setUIError('Password is invalid'));
      return false;
    }

    dispatch(removeUIError());
    return true;
  };

  return (
    <>
      <h3 className="auth__title">Register</h3>

      <form action="post" onSubmit={handleRegister}>
        {msgError && <div className="auth__alert-error">{msgError}</div>}

        <input
          className="auth__input"
          type="text"
          placeholder="Your name"
          name="userName"
          autoComplete="off"
          value={userName}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          type="text"
          placeholder="Your email"
          name="email"
          autoComplete="off"
          value={email}
          onChange={handleInputChange}
        />

        <input
          className="auth__input"
          type="password"
          name="password"
          placeholder="Your password"
          value={password}
          onChange={handleInputChange}
        />

        <input
          className="auth__input"
          type="password"
          name="confirmPassword"
          placeholder="Confirm password"
          value={confirmPassword}
          onChange={handleInputChange}
        />

        <button className="btn btn-primary btn-block mb-5" type="submit">
          Create Account
        </button>

        <Link className="link mb-5 mt-5" to="/auth/register">
          Already have an account?
        </Link>
      </form>
    </>
  );
};

export default RegisterScreen;
