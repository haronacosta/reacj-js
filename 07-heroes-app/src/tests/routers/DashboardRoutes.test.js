import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../auth/authContext';
import DashboardRoutes from '../../routers/DashboardRoutes';

describe('pruebas en el <DashboardRoutes />', () => {
  const contextValue = {
    user: {
      logged: true,
      name: 'Juan Perez',
    },
  };

  test('Debe mostrarse correctamente', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter initialEntries={['/']}>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text().trim()).toBe('Marvel Screen');
  });

  test('Debe mostrarse correctamente los heroes de DC', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter initialEntries={['/dc']}>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text().trim()).toBe('DCScreen');
  });
});
