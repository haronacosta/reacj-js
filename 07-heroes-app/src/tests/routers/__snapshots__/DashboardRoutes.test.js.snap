// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el <DashboardRoutes /> Debe mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-link nav-item text-info"
        >
          Juan Perez
        </span>
        <button
          className="btn nav-item nav-link"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        Marvel Screen
      </h1>
      Array [
        <h2>
          Hero List
        </h2>,
        <p
          className="text-warning"
        >
          Publisher: 
          Marvel Comics
        </p>,
        <div
          className="row row-cols-1 row-cols-md-2 g-4 animate__animated animate__fadeIn"
        >
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Spider Man"
                    className="card-img-top"
                    src="/assets/heroes/marvel-spider.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Spider Man
                  </div>
                  <p
                    className="card-text"
                  >
                    Peter Parker
                  </p>
                  <p
                    className="card-text"
                  >
                    Amazing Fantasy #15
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/marvel-spider"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Captain America"
                    className="card-img-top"
                    src="/assets/heroes/marvel-captain.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Captain America
                  </div>
                  <p
                    className="card-text"
                  >
                    Steve Rogers
                  </p>
                  <p
                    className="card-text"
                  >
                    Captain America Comics #1
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/marvel-captain"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Iron Man"
                    className="card-img-top"
                    src="/assets/heroes/marvel-iron.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Iron Man
                  </div>
                  <p
                    className="card-text"
                  >
                    Tony Stark
                  </p>
                  <p
                    className="card-text"
                  >
                    Tales of Suspense #39
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/marvel-iron"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Thor"
                    className="card-img-top"
                    src="/assets/heroes/marvel-thor.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Thor
                  </div>
                  <p
                    className="card-text"
                  >
                    Thor Odinson
                  </p>
                  <p
                    className="card-text"
                  >
                    Journey into Myster #83
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/marvel-thor"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Hulk"
                    className="card-img-top"
                    src="/assets/heroes/marvel-hulk.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Hulk
                  </div>
                  <p
                    className="card-text"
                  >
                    Bruce Banner
                  </p>
                  <p
                    className="card-text"
                  >
                    The Incredible Hulk #1
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/marvel-hulk"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Wolverine"
                    className="card-img-top"
                    src="/assets/heroes/marvel-wolverine.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Wolverine
                  </div>
                  <p
                    className="card-text"
                  >
                    James Howlett
                  </p>
                  <p
                    className="card-text"
                  >
                    The Incredible Hulk #180
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/marvel-wolverine"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Daredevil"
                    className="card-img-top"
                    src="/assets/heroes/marvel-daredevil.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Daredevil
                  </div>
                  <p
                    className="card-text"
                  >
                    Matthew Michael Murdock
                  </p>
                  <p
                    className="card-text"
                  >
                    Daredevil #1
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/marvel-daredevil"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Hawkeye"
                    className="card-img-top"
                    src="/assets/heroes/marvel-hawkeye.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Hawkeye
                  </div>
                  <p
                    className="card-text"
                  >
                    Clinton Francis Barton
                  </p>
                  <p
                    className="card-text"
                  >
                    Tales of Suspense #57
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/marvel-hawkeye"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Cyclops"
                    className="card-img-top"
                    src="/assets/heroes/marvel-cyclops.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Cyclops
                  </div>
                  <p
                    className="card-text"
                  >
                    Scott Summers
                  </p>
                  <p
                    className="card-text"
                  >
                    X-Men #1
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/marvel-cyclops"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Silver Surfer"
                    className="card-img-top"
                    src="/assets/heroes/marvel-silver.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Silver Surfer
                  </div>
                  <p
                    className="card-text"
                  >
                    Norrin Radd
                  </p>
                  <p
                    className="card-text"
                  >
                    The Fantastic Four #48
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/marvel-silver"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>,
      ]
    </div>
  </div>,
]
`;

exports[`pruebas en el <DashboardRoutes /> Debe mostrarse correctamente los heroes de DC 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-link nav-item text-info"
        >
          Juan Perez
        </span>
        <button
          className="btn nav-item nav-link"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        DCScreen
      </h1>
      Array [
        <h2>
          Hero List
        </h2>,
        <p
          className="text-warning"
        >
          Publisher: 
          DC Comics
        </p>,
        <div
          className="row row-cols-1 row-cols-md-2 g-4 animate__animated animate__fadeIn"
        >
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Batman"
                    className="card-img-top"
                    src="/assets/heroes/dc-batman.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Batman
                  </div>
                  <p
                    className="card-text"
                  >
                    Bruce Wayne
                  </p>
                  <p
                    className="card-text"
                  >
                    Detective Comics #27
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/dc-batman"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Superman"
                    className="card-img-top"
                    src="/assets/heroes/dc-superman.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Superman
                  </div>
                  <p
                    className="card-text"
                  >
                    Kal-El
                  </p>
                  <p
                    className="card-text"
                  >
                    Action Comics #1
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/dc-superman"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Flash"
                    className="card-img-top"
                    src="/assets/heroes/dc-flash.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Flash
                  </div>
                  <p
                    className="card-text"
                  >
                    Jay Garrick
                  </p>
                  <p
                    className="text-muted"
                  >
                    Jay Garrick, Barry Allen, Wally West, Bart Allen
                  </p>
                  <p
                    className="card-text"
                  >
                    Flash Comics #1
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/dc-flash"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Green Lantern"
                    className="card-img-top"
                    src="/assets/heroes/dc-green.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Green Lantern
                  </div>
                  <p
                    className="card-text"
                  >
                    Alan Scott
                  </p>
                  <p
                    className="text-muted"
                  >
                    Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
                  </p>
                  <p
                    className="card-text"
                  >
                    All-American Comics #16
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/dc-green"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Green Arrow"
                    className="card-img-top"
                    src="/assets/heroes/dc-arrow.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Green Arrow
                  </div>
                  <p
                    className="card-text"
                  >
                    Oliver Queen
                  </p>
                  <p
                    className="card-text"
                  >
                    More Fun Comics #73
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/dc-arrow"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Wonder Woman"
                    className="card-img-top"
                    src="/assets/heroes/dc-wonder.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Wonder Woman
                  </div>
                  <p
                    className="card-text"
                  >
                    Princess Diana
                  </p>
                  <p
                    className="card-text"
                  >
                    All Star Comics #8
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/dc-wonder"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Martian Manhunter"
                    className="card-img-top"
                    src="/assets/heroes/dc-martian.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Martian Manhunter
                  </div>
                  <p
                    className="card-text"
                  >
                    J'onn J'onzz
                  </p>
                  <p
                    className="text-muted"
                  >
                    Martian Manhunter
                  </p>
                  <p
                    className="card-text"
                  >
                    Detective Comics #225
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/dc-martian"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Robin/Nightwing"
                    className="card-img-top"
                    src="/assets/heroes/dc-robin.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Robin/Nightwing
                  </div>
                  <p
                    className="card-text"
                  >
                    Dick Grayson
                  </p>
                  <p
                    className="card-text"
                  >
                    Detective Comics #38
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/dc-robin"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Blue Beetle"
                    className="card-img-top"
                    src="/assets/heroes/dc-blue.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Blue Beetle
                  </div>
                  <p
                    className="card-text"
                  >
                    Dan Garret
                  </p>
                  <p
                    className="text-muted"
                  >
                    Dan Garret, Ted Kord, Jaime Reyes
                  </p>
                  <p
                    className="card-text"
                  >
                    Mystery Men Comics #1
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/dc-blue"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-4"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Black Canary"
                    className="card-img-top"
                    src="/assets/heroes/dc-black.jpg"
                  />
                </div>
                <div
                  className="col-8"
                >
                  <div
                    className="card-title"
                  >
                    Black Canary
                  </div>
                  <p
                    className="card-text"
                  >
                    Dinah Drake
                  </p>
                  <p
                    className="text-muted"
                  >
                    Dinah Drake, Dinah Lance
                  </p>
                  <p
                    className="card-text"
                  >
                    Flash Comics #86
                  </p>
                  <a
                    className="pl-4 btn btn-outline-success"
                    href="/hero/dc-black"
                    onClick={[Function]}
                  >
                    Más info
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>,
      ]
    </div>
  </div>,
]
`;
